server/src/main/java/com/google/musicanalysis/site/MusixRequest.java

```diff
@@ -0,0 +1,75 @@
+package com.google.musicanalysis.site;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+/*
+    Handles gathering data from the Musixmatch API.
+    This is structured the way it is becuase I will be adding more
+    to each function.
+*/
+public class MusixRequest {
+
+  // Url building
+  private static final String BASE_URL = "https://api.musixmatch.com/ws/1.1/";
+  private String operation;
+  private static final String OUTPUT_REQS = "?format=jsonp&callback=callback&";
+  private String parameters;
+  private static final String apiKey = "&apikey=";
+
+  /**
+   * Constructor.
+   *
+   * @param operation the API method to send to Musixmatch
+   * @param parameters the parameter string to be sent to Musixmatch
+   */
+  public MusixRequest(String operation, String parameters) {
+    this.operation = operation;
+    this.parameters = parameters;
+  }
+
+  /** @return the result from Musixmatch API as string */
+  public String grabResponse() throws MalformedURLException, IOException {
```

Comment:
Please explain why you are wrapping `getResult()` with `grabResponse()`. Are you planning to add more stuff to this method later?

Reply:
I did above. 
Copied:
I put this here to make it easiest on the readers as possible. When they scan through the code, they see that in order to get the response from the Musixmatch API, they instantiate a new class and run obj.grabResponse(), which will return as string. All of the work will remain in private functions to give an "under the hood effect". If they're interested in what's going on, they can look for more info, if not, they can just use it.

I am also planning to add more stuff later, as mentioned in the implementation notes and in the comments at the top of the file.

Reply:
+1 to Ibiyemi. You should just make `getResult` public and remove `grabResponse`. Otherwise this obfuscation hurts readability more than improves it.